python/GafferSceneTest/MeshSplitTest.py

```diff
@@ -0,0 +1,293 @@
+		]
+
+		self.assertEqual( self.sceneSummary( MeshSplit["out"] ), splitBothResult )
+
+		MeshSplit["nameFromSegment"].setValue( True )
+
+		splitBothResultWithNames = [
+			"/ NullObject",
+			"/object NullObject",
+			"/object/b 156",
+			"/object/c 125",
+			"/object/a 119",
+			"/noPrimVars 800",
+			"/objectButNotReally Camera",
+			"/object1 NullObject",
+			"/object1/1 800",
+			"/object1/b 800",
+			"/object1/object 400",
+			"/object1/b1 156",
+			"/object1/c 125",
+			"/object1/a 119",
+		]
+
+		self.assertEqual( self.sceneSummary( MeshSplit["out"] ), splitBothResultWithNames )
+
+		# If we include a mesh without the primvar in our filter, we get an error
+		MeshSplit["nameFromSegment"].setValue( False )
+		Filter["paths"].setValue( IECore.StringVectorData( [ '/*' ] ) )
+		with six.assertRaisesRegex( self, RuntimeError, 'Cannot find primitive variable "uniformString"' ) :
+			self.sceneSummary( MeshSplit["out"] )
+
+		Filter["paths"].setValue( IECore.StringVectorData( [ '/object*' ] ) )
+
+		self.assertEqual( self.sceneSummary( MeshSplit["out"] ), splitBothResult )
+
+		Filter["paths"].setValue( IECore.StringVectorData( [ '/object1', '/object1/object' ] ) )
+
+		splitNestedResult = [
+			"/ NullObject",
+			"/object 400",
+			"/noPrimVars 800",
+			"/objectButNotReally Camera",
+			"/object1 NullObject",
+			"/object1/1 800",
+			"/object1/b 800",
+			"/object1/object NullObject",
+			"/object1/object/0 156",
+			"/object1/object/1 119",
+			"/object1/object/2 125",
+			"/object1/0 156",
+			"/object1/11 119",
+			"/object1/2 125",
+		]
+
+		self.assertEqual( self.sceneSummary( MeshSplit["out"] ), splitNestedResult )
+	
+		# OK, now lets stop mucking about with weird hierarchies and just test a couple of different primvars	
+		Filter["paths"].setValue( IECore.StringVectorData( [ '/object' ] ) )
+
+		MeshSplit["segment"].setValue( 'uniformThickRings' )
+
+		splitRingsResult = [
+			"/ NullObject",
+			"/object NullObject",
+			"/object/0 40",
+			"/object/1 40",
+			"/object/2 40",
+			"/object/3 40",
+			"/object/4 40",
+			"/object/5 40",
+			"/object/6 40",
+			"/object/7 40",
+			"/object/8 40",
+			"/object/9 40",
+			"/noPrimVars 800",
+			"/objectButNotReally Camera",
+			"/object1 400",
+			"/object1/1 800",
+			"/object1/b 800",
+			"/object1/object 400"
+		]
+
+		self.assertEqual( self.sceneSummary( MeshSplit["out"] ), splitRingsResult )
+
+		MeshSplit["segment"].setValue( 'uniformIndividualFaces' )
+		self.assertEqual(
+			MeshSplit['out'].childNames( "/object" ),
+			IECore.InternedStringVectorData( [ str( i ) for i in range( 400 ) ] )
+		)
+		for i in range( 400 ):
+			self.assertEqual(
+				len( MeshSplit['out'].object( "/object/%i" % i ).verticesPerFace ),
+				1
+			)
+
+		MeshSplit["nameFromSegment"].setValue( True )
+		with six.assertRaisesRegex( self, RuntimeError, "Cannot use nameFromSegment with primitive variable uniformIndividualFaces because it is not a string." ) :
+			self.sceneSummary( MeshSplit["out"] )
+			
+
```

Comment:
Would be good to sprinkle `assertSceneValid( meshSplit["out"] )` through this function to get some coverage for the set and bound computations for free.